src/library.js

```diff
@@ -2214,15 +2287,34 @@ const createPilingJs = (rootElement, initOptions = {}) => {
     const updatedItems = [];
     const updatedPileItems = [];
 
+    if (state.items !== newState.items) {
+      if (state.items.length) {
+        const newItems = [];
+        newState.items.forEach((item, id) => {
+          if (item.src !== state.items[id].src) {
```

Comment:
The `if` statement can be added as a `.filter()` after `.items`. That would make this code a little clearer

Reply:
Hmm if we use
```
newState.items
  .filter((item, id) => item.src !== state.items[id].src)
  .forEach((item, id) => {
    newItems.push({
      id,
      src: item.src
    });
```
the `id` in `forEach` would be wrong. Although we can use `indexOf(item)` to find the index. Is it necessary?

Reply:
You're absolutely right! I remember I implemented it with `filter` and immediately reverted the code. Can you do me a favor and add a comment before the `if` statement? So next time we look at the code it's clear why we can't use `filter`? Thanks.